src/legacy/core_plugins/vis_type_timelion/public/helpers/panel_utils.ts

```diff
@@ -0,0 +1,229 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { cloneDeep, defaults, merge } from 'lodash';
+import moment, { Moment } from 'moment-timezone';
+
+import { TimefilterContract } from 'src/plugins/data/public';
+import { IUiSettingsClient } from 'kibana/public';
+
+// @ts-ignore
+import { calculateInterval, DEFAULT_TIME_FORMAT } from '../../../timelion/common/lib';
+import { tickFormatters } from './tick_formatters';
+import { xaxisFormatterProvider } from './xaxis_formatter';
+import { generateTicksProvider } from './tick_generator';
+import { Series } from './timelion_request_handler';
+
+export interface Axis {
+  delta?: number;
+  max?: number;
+  min?: number;
+  mode: string;
+  options?: {
+    units: { prefix: string; suffix: string };
+  };
+  tickSize?: number;
+  ticks: number;
+  tickLength: number;
+  timezone: string;
+  tickDecimals?: number;
+  tickFormatter: ((val: number) => string) | ((val: number, axis: Axis) => string);
+  tickGenerator?(axis: Axis): number[];
+  units?: { type: string };
+}
+interface IOptions {
```

Comment:
If you install the types, it seems like this interface is not necessary anymore because it's covered by `jquery.flot.plotOptions`

Reply:
Thanks.
Removed.